pkg/file/joiner/joiner.go

```diff
@@ -406,6 +406,12 @@ func (j *joiner) processChunkAddresses(ctx context.Context, fn swarm.AddressIter
 			if j.refLength == encryption.ReferenceSize && i < shardCnt {
 				addr = swarm.NewAddress(data[cursor : cursor+swarm.HashSize*2])
 			}
+
+			// not a shard
+			if i >= shardCnt {
```

Comment:
maybe it should be `i >shardCnt` 🤔 
you are also reading `i` from a goroutine, there could be some data race issues here. Pls also check if we are accessing variables here that we should not be without some synchronization mechanism.

Reply:
`i` is 0-indexed. So it should be `i >= shardCnt`
The gourtine that reads `i` has to return before the next iteration. See the waitgroup used below. So there can be no race here.